Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -1687,7 +1695,46 @@ private void LogCommandLineTooLong(SandboxedProcessInfo info)
                 allReportedFileAccesses: allFileAccesses,
                 detouringStatuses: result.DetouringStatuses,
                 maxDetoursHeapSize: maxDetoursHeapSize,
-                containerConfiguration: m_containerConfiguration);
+                containerConfiguration: m_containerConfiguration,
+                pipProperties: pipProperties);
+        }
+
+        private async Task<Dictionary<string, int>> UpdatePipPropertiesCountsAsync(SandboxedProcessResult result, Dictionary<string, int> pipProperties)
+        {
+            const string PipPropertyPrefix = "PipProperty_";
+
+            Regex pipPropertyRegex = new Regex(PipPropertyPrefix + @"(?<" + PipPropertyPrefix + @">.+?)_EndProperty", RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.Multiline);
+            OutputFilter propertyFilter = new OutputFilter(pipPropertyRegex, PipPropertyPrefix);
+
+            string errorMatches = await TryFilterAsync(result.StandardError, propertyFilter, appendNewLine: true);
+            string outputMatches = await TryFilterAsync(result.StandardOutput, propertyFilter, appendNewLine: true);
+            string allMatches = errorMatches + Environment.NewLine + outputMatches;
+
+            if (!string.IsNullOrWhiteSpace(allMatches))
+            {
+                string[] matchedProperties = allMatches.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+                if (pipProperties == null)
+                {
+                    pipProperties = matchedProperties.ToDictionary(val => val, val => 1);
+                }
+                else
+                {
+                    foreach (string property in matchedProperties)
+                    {
+                        if (pipProperties.ContainsKey(property))
+                        {
+                            pipProperties[property] = ++pipProperties[property];

```

Comment:
>pipProperties[property] = ++pipProperties[property]; [](start = 28, length = 52)

Is `++pipProperties[property]` alone not sufficient? I looks funny to have the assignment and the `++` operator.